File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.js
function solution(n) {
	let i = 1;
    let mul = 1;
    while (i <= n) {
    	// i가 1일 때부터 순차적으로 값을 곱한다.
        i++;
        mul *= i;
        // mul값이 n보다 커졌을 경우, i-1 값을 리턴한다.
        if (mul > n) return i - 1;
        // mul값이 n과 같을 경우, i값을 리턴한다.
        if (mul === n) return i;
	}
}

File: 프로그래머스/0/120848. 팩토리얼/팩토리얼.py
import math
box = []

def solution(n):
    for i in range(1,11):
        answer = math.factorial(i)
        box.append(answer)
        
        if box[i-2] < n < box[i-1]:
            return i-1
        elif box[i-2] <= n <= box[i-1]:
            return i